cli/src/main/java/org/jboss/galleon/cli/cmd/HelpCommand.java

```diff
@@ -120,17 +125,53 @@ protected void runCommand(PmCommandInvocation session) throws CommandExecutionEx
                 builder.append(str).append(" ");
             }
             String cmd = builder.toString().trim();
-            try {
-                CommandContainer<?, ?> container = registry.getCommand(command.get(0), null);
-                if (command.size() > 1) {
-                    if (container.getParser().getChildParser(command.get(1)) == null) {
+            if (!"all".equalsIgnoreCase(cmd)) {
+                try {
+                    CommandContainer<?, ?> container = registry.getCommand(command.get(0), null);
+                    if (command.size() > 1) {
+                        if (container.getParser().getChildParser(command.get(1)) == null) {
+                            throw new CommandExecutionException(CliErrors.commandNotFound(cmd));
+                        }
+                    }
+                } catch (CommandNotFoundException ex) {
+                    throw new CommandExecutionException(CliErrors.commandNotFound(cmd));
+                }
+                session.println(session.getHelpInfo(cmd));
+            } else if ("all".equalsIgnoreCase(cmd)) {
+                Set<String> allCommandNames = registry.getAllCommandNames();
+                Map<String, Set<String>> groupedCommands = new TreeMap<>();
+                for (String commandName : allCommandNames) {
+                    try {
+                        String group = CommandDomain.getDomain(registry.getCommand(commandName, null).getParser().getCommand()).getDescription();
+                        String commandTree = HelpSupport.getCommandTree(registry, commandName);
+                        Set<String> currentDescriptions = new TreeSet<>();
+
+                        if (group != null && "".equals(group)) {
```

Comment:
CommandDomain.getDomain returns OTHER by default, I don't see why you need to retrieve the domain of this command. You should modify the HelpSupport.addCommandDomains to benefit from Domain handling.